client/client.go

```diff
@@ -91,38 +111,39 @@ func (q Query) ClientKey() string {
 	return ""
 }
 
-func (q Query) ClientCertRequired() bool {
-	val, ok := q[""]
-	return ok && val.(bool)
-}
-
+// Insecure returns whether the query has a field called "insecure" whose value is true.
 func (q Query) Insecure() bool {
 	val, ok := q["insecure"]
 	return ok && val.(bool)
 }
 
+// SNI returns whether the query has a field called "sni" whose value is true.
 func (q Query) SNI() bool {
 	val, ok := q["sni"]
 	return ok && val.(bool)
 }
 
+// IsWebsocket returns whether the query's URL starts with "ws:".
 func (q Query) IsWebsocket() bool {
-	return strings.HasPrefix(q.Url(), "ws:")
+	return strings.HasPrefix(q.URL(), "ws:")
 }
 
-func (q Query) Url() string {
+// URL returns the query's URL.
+func (q Query) URL() string {
 	return q["url"].(string)
 }
 
+// Method returns the query's method or "GET" if unspecified.
 func (q Query) Method() string {
 	val, ok := q["method"]
 	if ok {
 		return val.(string)
-	} else {
-		return "GET"
 	}
+	return "GET"
```

Comment:
Log that method is returning a default value.

Reply:
Are you sure? I don't agree. The log output exists to help us debug the code; all real errors are recorded in `query.result.error`. I don't see how cluttering that output with stuff like `Query does not include a method; returning the default GET` will help anyone debug anything in most cases.

Reply:
Not really, you are returning a value that could be evaluated to true in the test, even if the result is not correct. I'm not sure that it is valid to just assume a GET method here. I think would be better to return an empty string or "UNKNOW METHOD". Something that tells the test that method is not correct.